src/tests/functional/plugin/shared/include/behavior/ov_infer_request/memory_states.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "base/behavior_test_utils.hpp"
+#include "common_test_utils/test_common.hpp"
+#include "functional_test_utils/ov_plugin_cache.hpp"
+#include "openvino/runtime/auto/properties.hpp"
+
+namespace ov {
+namespace test {
+namespace behavior {
+
+using memoryStateParams = std::tuple<std::shared_ptr<ov::Model>,  // Model to work with
+                                     std::vector<std::string>,    //  Memory States to query
+                                     std::string,                 // Target device name
+                                     ov::AnyMap>;                 // device configuration
+
+class OVInferRequestVariableStateTest : public testing::WithParamInterface<memoryStateParams>,
```

Comment:
@olpipi please, note that `InferRequestQueryStateExceptionTest` is used in template only and not ported in scope of this PR